src/components/context/FetchDataContext.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useContext, createContext, useEffect, useCallback } from "react";
+import { usePagination } from "../dataTable/pagination/PaginationContext";
+import { BASE_URL } from "../../../config";
+
+interface FetchDataContextType {
+  datasets: any[];
+  searchDatasets: any[];
+  facets: any[];
+  urlSearchParams: string | null;
+  urlSelectedFilters: any[];
+  setUrlSearchParams: React.Dispatch<React.SetStateAction<string | null>>;
+  setUrlSelectedFilters: React.Dispatch<React.SetStateAction<any[]>>;
+  fetchDatasets: () => Promise<void>;
+  fetchSearchDatasets: () => Promise<void>;
+  fetchCount: () => Promise<void>;
+  fetchFacets: () => Promise<void>;
+}
+
+const FetchDataContext = createContext<FetchDataContextType | undefined>(undefined);
+
+export const useFetchData = () => {
+  const context = useContext(FetchDataContext);
+  if (!context) {
+    throw new Error('useFetchData must be used within a FetchDataProvider');
+  }
+  return context;
+};
+
+const tokenObj = {
+  method: 'GET',
+  headers: {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json',
+    'Authorization': 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI2NWE3YWRhYzQwM2IzZGViZDE0NzNmNzYiLCJ1c2VybmFtZSI6ImFkbWluIiwiZW1haWwiOiJhZG1pbkB5b3VyLnNpdGUiLCJhdXRoU3RyYXRlZ3kiOiJsb2NhbCIsIl9fdiI6MCwiaWQiOiI2NWE3YWRhYzQwM2IzZGViZDE0NzNmNzYiLCJpYXQiOjE3MDYyNjA2NjgsImV4cCI6MTcwNjI2NDI2OH0.RZwXIj2tIX2mH8DET9WfpR5ijU7IVvk44u7uF71yiFc'
+  }
+};
+
+const buildUrl = (endpoint: string, queryParams?: string) => {
+  let url = `${BASE_URL}${endpoint}`
+  if (queryParams) {
+    url += `${queryParams}`
+  }
+  return url
+}
+
+export const FetchDataProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+  const [datasets, setDatasets] = useState<any[]>([]);
+  const [facets, setFacets] = useState<any[]>([]);
+  const [searchDatasets, setSearchDatasets] = useState<any[]>([]);
+
+  const [urlSearchParams, setUrlSearchParams] = useState<string | null>(null);
```

Comment:
Wouldn't it better to set default value to empty string? so you can remove the null type